### PR TITLE
Iabthao

### PR DESCRIPTION
**Pull request by iabthao**

### CODE REVIEW

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
+package edu.metrostate.cardealer.functionality;
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: Please pull master into your branch and move Vehicle_ImportFile.java and Vehicle_ViewList.java into the activity directory. Please also rename them to include the word Activity within the class/file name

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+public class Vehicle_ViewList extends AppCompatActivity {
+    ListView vehicle_list;
+    String[] arr = {"Null", "Data"};
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_vehicle_view_list);
+
+        Bundle bundle = getIntent().getExtras();
+        ArrayList<Vehicle> vehiclelist = bundle.getParcelableArrayList("vehiclelist_data");
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java]: You can change the way vehicles are accessed through calling StateManager.dealerGroup.getAllVehicles(). I added the getAllVehicles() method in my PR if you want to look at it

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        ArrayAdapter<Vehicle> vehicleArray;
+        vehicleArray
+                = new ArrayAdapter<Vehicle>(
+                this,
+                androidx.appcompat.R.layout.support_simple_spinner_dropdown_item,
+               vehiclelist);
+        vehicle_list.setAdapter(vehicleArray);
+        }
+    }
+    }
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java]: Maybe add a way for if you click on a vehicle it brings you to Mike's Vehicle page where you can edit a specific vehicle. I don't know what he has called the class so we can talk about this in class

File: app/src/main/java/edu/metrostate/cardealer/inventory/Vehicle.java [hunk 4]
```diff
@@ -1,8 +1,11 @@
 package edu.metrostate.cardealer.inventory;
 
+import android.os.Parcel;
+import android.os.Parcelable;
+
 import java.util.Date;
 
-public class Vehicle {
+public class Vehicle implements Parcelable {
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/inventory/Vehicle.java]: Through accessing StateManager.dealerGroup.getAllVehicles() directly we can just get rid of the changes in this file

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_vehicle_import_file);
+
+        Button send_button= findViewById(R.id.send_data);
+        send_button.setOnClickListener(v -> {
+            Intent intent = new Intent(this, Vehicle_ViewList.class);
+
+
+            Bundle bundle = new Bundle();
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: Bundle passing to the vehicle list activity isn't needed because of the global StateManager.dealerGroup is accessible from every class

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+                        File file = null;
+                        if (uri != null) {
+                            file = new File(uri.getPath());
+                        }
+                        if (file != null) {
+                            path = file.getAbsolutePath();
+                        }
+                        if(path.substring(path.lastIndexOf(".") + 1, path.length()).equals("xml"))
+                        {
+                            final TextView xmlField = findViewById(R.id.xml_path);
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: make sure to call the XML parser and add the dealers to the StateManager.dealerGroup, and display any dealers that could not add any new vehicles

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+                        }
+
+                        if (path != null) {
+                            if(path.substring(path.lastIndexOf(".") + 1, path.length()).equals("json"))
+                            {
+                                final TextView jsonField = findViewById(R.id.json_path);
+                                jsonField.setText(path);
+
+                                vehicles = (ArrayList<Vehicle>) VehicleJSONParser.read(file);
+                            }else{
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: Make sure to add the vehicles to the StateManager.dealerGroup and then display which dealers could not add any new vehicles. (Acquisition disabled)

Comment (tomatogrove) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: This message should be a dialog. You can have a success dialog that shows which dealers weren't enabled for vehicle acquisition, if any.

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+                            xmlField.setText(path);
+                        }else{
+                            final TextView errorField = findViewById(R.id.error_message);
+                            errorField.setText("Wrong file format");
+                        }
+                    }
+
+                }
+            }
+    );
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: Please include 

    @Override
    protected void onPause() {
        super.onPause();

        StateManager.save();
    }

within this file. It makes sure that state is saved whenever the Activity is paused.

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+        Bundle bundle = getIntent().getExtras();
+        ArrayList<Vehicle> vehiclelist = bundle.getParcelableArrayList("vehiclelist_data");
+        if(vehiclelist.size()==0){
+            vehicle_list = findViewById(R.id.vehicle_list);
+            ArrayAdapter<String> vehicleArray;
+            vehicleArray
+                    = new ArrayAdapter<String>(
+                    this,
+                    androidx.appcompat.R.layout.support_simple_spinner_dropdown_item,
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java]: in class could you show me the use of the spinner?

File: app/src/main/java/edu/metrostate/cardealer/CarDealerApplication.java [hunk 10]
```diff
@@ -9,46 +9,40 @@
 import java.util.ArrayList;
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/CarDealerApplication.java]: please revert the changes youve made to this file and pull master

File: app/src/main/java/edu/metrostate/cardealer/inventory/DealerGroup.java [hunk 11]
```diff
@@ -1,5 +1,7 @@
 package edu.metrostate.cardealer.inventory;
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/inventory/DealerGroup.java]: please revert the changes youve made to this file and pull master

File: app/src/main/java/edu/metrostate/cardealer/storage/StateManager.java [hunk 12]
```diff
@@ -4,18 +4,21 @@
 import edu.metrostate.cardealer.inventory.DealerGroup;
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/storage/StateManager.java:4]: please revert the changes youve made to this file and pull master

File: app/src/main/java/edu/metrostate/cardealer/functionality/VehicleJSONParser.java [hunk 13]
```diff
@@ -2,12 +2,15 @@
 
 import java.io.File;
 import java.io.IOException;
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/functionality/VehicleJSONParser.java]: please revert the changes youve made to this file and pull master

File: app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+            }
+    );
+    public void addVehicle(){
+        //need to change to Mike class
+        Intent intent = new Intent(this, Vehicle_ViewListActivity.class);
+        startActivity(intent);
+    }
+    @Override
+    protected void onPause() {
+        app = (CarDealerApplication) getApplication();
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java]: you can remove the 

app = (CarDealerApplication) getApplication();

same with the CarDealerApplication variable at the top of the class.

File: app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+                            errorField.setText("Wrong file format");
+                        }
+                    }
+
+                }
+            }
+    );
+    public void addVehicle(){
+        //need to change to Mike class
+        Intent intent = new Intent(this, Vehicle_ViewListActivity.class);
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java]: This should be going to my class NewVehicleFormActivity. Its within master now so it shouldnt be giving any errors

File: app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java [hunk 16]
```diff
@@ -0,0 +1,193 @@
...
+import edu.metrostate.cardealer.functionality.VehicleXMLParser;
+import edu.metrostate.cardealer.inventory.Dealership;
+import edu.metrostate.cardealer.inventory.Vehicle;
+import edu.metrostate.cardealer.storage.StateManager;
+
+public class Vehicle_ImportFileActivity extends AppCompatActivity {
+    String path;
+    List<Dealership> vehicles;
+    List<Vehicle> vehicleJson;
+    private CarDealerApplication app;
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java]: app variable not necessary

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java (hunk 1 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: Completed

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java (hunk 2 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java]: Completed

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java (hunk 3 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java]: Not sure what this is "Maybe add a way for if you click on a vehicle it brings you to Mike's Vehicle page where you can edit a specific vehicle. I don't know what he has called the class so we can talk about this in class."

File: app/src/main/java/edu/metrostate/cardealer/inventory/Vehicle.java (hunk 4 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/inventory/Vehicle.java]: Completed.

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java (hunk 5 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: Completed.

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java (hunk 6 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: Completed. 
Not sure what this is "display any dealers that could not add any new vehicles." Please redirect to the right person.

File: app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java [hunk 17]
```diff
@@ -0,0 +1,205 @@
...
+
+                        if (path != null) {
+                            if(path.substring(path.lastIndexOf(".") + 1, path.length()).equals("json"))
+                            {
+                                final TextView jsonField = findViewById(R.id.json_path);
+                                jsonField.setText(path);
+
+                                vehicleJson = (ArrayList<Vehicle>) VehicleJSONParser.read(file);
+                                for(Vehicle vehicle : vehicleJson){
+    System.out.println(vehicle.toString());
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java]: could you remove this println

File: app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java [hunk 18]
```diff
@@ -0,0 +1,205 @@
...
+                            path = file.getAbsolutePath();
+                        }
+
+                        if (path != null) {
+                            if(path.substring(path.lastIndexOf(".") + 1, path.length()).equals("json"))
+                            {
+                                final TextView jsonField = findViewById(R.id.json_path);
+                                jsonField.setText(path);
+
+                                vehicleJson = (ArrayList<Vehicle>) VehicleJSONParser.read(file);
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java]: this cast isn't necessary because read already returns an List of vehicles

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java (hunk 7 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: Completed. Not sure what this is " display which dealers could not add any new vehicles. (Acquisition disabled)."

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java (hunk 9 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java]: Completed.

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java (hunk 8 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: Completed.

File: app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java [hunk 19]
```diff
@@ -0,0 +1,205 @@
...
+                        File file = null;
+                        if (uri != null) {
+                            file = new File(uri.getPath());
+                        }
+                        if (file != null) {
+                            path = file.getAbsolutePath();
+                        }
+                        if(path.substring(path.lastIndexOf(".") + 1, path.length()).equals("xml"))
+                        {   vehicles = VehicleXMLParser.read(file);
+
```
Comment (tomatogrove) [inline app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java]: could you change the vehicles variable name and then also add

`StateManager.dealerGroup.addIncomingDealers(dealers)`

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java (hunk 3 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java]: I added a button add vehicle on my main frame. Please connect with Mike's Vehicle page.

Comment (tomatogrove) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ViewList.java]: a button with an intent to his class

File: app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java (hunk 7 above)
Comment (tomatogrove) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: https://github.com/tomatogrove/ICS372-Assignment3/blob/master/app/src/main/java/edu/metrostate/cardealer/activity/VehicleListActivity.java#L57-L68

In the VehicleListActivity within main there is an example of a Dialog. 

this is what a dialog looks like on an android screen. 
https://i.stack.imgur.com/K9rRe.png

Comment (tomatogrove) [reply app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java]: Basically the purpose of this is to show the user what has happened to their file, because what the user sees is that their file is on the screen but not that anything has happened to it. The dialog is there to communicate with the user what has happened.

File: app/src/main/java/edu/metrostate/cardealer/storage/StateManager.java (hunk 12 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/storage/StateManager.java:4]: Completed

File: app/src/main/java/edu/metrostate/cardealer/functionality/VehicleJSONParser.java (hunk 13 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/functionality/VehicleJSONParser.java]: Completed

File: app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java (hunk 14 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java]: Completed

File: app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java (hunk 15 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java]: Completed

File: app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java (hunk 16 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java]: Completed

File: app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java (hunk 17 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java]: Already removed.

File: app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java (hunk 18 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java]: Completed.

File: app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java (hunk 19 above)
Comment (iabthao) [reply app/src/main/java/edu/metrostate/cardealer/activity/Vehicle_ImportFileActivity.java]: Already completed before.

**Review by tomatogrove — APPROVED** [review]
